packages/grafana-data/src/types/datasource.ts

```diff
@@ -494,6 +494,10 @@ export interface DataQueryError {
 export interface DataQueryRequest<TQuery extends DataQuery = DataQuery> {
   requestId: string; // Used to identify results and optionally cancel the request in backendSrv
 
+  // adds requestId querystring param to fetch URL so it can be correlated in an async PerformanceObserver
+  // used for tracking frontend perf metrics, like response sizes, timings, gzip compression, etc.
+  appendRequestId?: boolean;
```

Comment:
🤔 i don't like that this is mixed between prometheus specific and has top-level interface changes.

for sure this should not be here -- i think we can add the datasource type to all datasource requests URL (it already has it in headers)